FreeRTOS-Plus/Demo/AWS/Device_Shadow_Windows_Simulator/Device_Shadow_Demo/DemoTasks/ShadowDemoMainExample.c

```diff
@@ -271,6 +327,126 @@ static void prvUpdateAcceptedHandler( MQTTPublishInfo_t * pxPublishInfo );
+
+    while( ( xDeleteResponseReceived != pdTRUE ) &&
+           ( ucCount++ < MQTT_PROCESS_LOOP_DELETE_RESPONSE_COUNT_MAX ) &&
+           ( xMQTTStatus == MQTTSuccess ) )
+    {
+        /* Event callback will set #xDeleteResponseReceived when receiving an
+         * incoming publish on either `/delete/accepted` or `/delete/rejected`
+         * Shadow topics. */
+        xMQTTStatus = MQTT_ProcessLoop( pxMQTTContext, MQTT_PROCESS_LOOP_TIMEOUT_MS );
+    }
+
+    if( ( xMQTTStatus != MQTTSuccess ) || ( xDeleteResponseReceived != pdTRUE ) )
+    {
+        LogError( ( "MQTT_ProcessLoop failed to receive a response for Shadow delete operation:"
+                    " LoopDuration=%u, MQTT Status=%s.",
+                    ( MQTT_PROCESS_LOOP_TIMEOUT_MS * ucCount ),
+                    MQTT_Status_strerror( xMQTTStatus ) ) );
+        xReturnStatus = pdFAIL;
+    }
+
+    return xReturnStatus;
+}
+
+/*-----------------------------------------------------------*/
+
+static void prvDeleteRejectedHandler( MQTTPublishInfo_t * pxPublishInfo )
+{
+    JSONStatus_t result = JSONSuccess;
+    char * pcOutValue = NULL;
+    uint32_t ulOutValueLength = 0UL;
+    uint32_t ulErrorCode = 0UL;
+
+    configASSERT( pxPublishInfo != NULL );
+    configASSERT( pxPublishInfo->pPayload != NULL );
+
+    LogInfo( ( "/delete/rejected json payload:%s.", ( const char * ) pxPublishInfo->pPayload ) );
+
+    /* The payload will look similar to this:
+     * {
+     *    "code": error-code,
+     *    "message": "error-message",
+     *    "timestamp": timestamp,
+     *    "clientToken": "token"
+     * }
+     */
+
+    /* Make sure the payload is a valid json document. */
+    result = JSON_Validate( pxPublishInfo->pPayload,
+                            pxPublishInfo->payloadLength );
+
+    if( result == JSONSuccess )
+    {
+        /* Then we start to get the version value by JSON keyword "version". */
+        result = JSON_Search( ( char * ) pxPublishInfo->pPayload,
+                              pxPublishInfo->payloadLength,
+                              SHADOW_DELETE_REJECTED_ERROR_CODE_KEY,
+                              SHADOW_DELETE_REJECTED_ERROR_CODE_KEY_LENGTH,
+                              &pcOutValue,
+                              ( size_t * ) &ulOutValueLength );
+    }
+    else
+    {
+        LogError( ( "The json document is invalid!!" ) );
+    }
+
+    if( result == JSONSuccess )
+    {
+        LogInfo( ( "Error code is: %.*s.",
+                   ulOutValueLength,
+                   pcOutValue ) );
+
+        /* Convert the extracted value to an unsigned integer value. */
+        ulErrorCode = ( uint32_t ) strtoul( pcOutValue, NULL, 10 );
+    }
+    else
+    {
+        LogError( ( "No error code in json document!!" ) );
+    }
+
+    LogInfo( ( "Error code:%lu.", ulErrorCode ) );
+
+    /* Mark Shadow delete operation as a success if error code is 404. */
+    if( ulErrorCode == 404 )
+    {
+        xShadowDeleted = pdTRUE;
+    }
```

Comment:
Seems excessive to me to convert this to an integer - wouldn't it be less work to check 3 characters?

Reply:
That's a good suggestion. Thanks